### PR TITLE
Add option to bypass the RPC layer for Java21 Runtime

### PR DESCRIPTION
**Pull request by lachlan-roberts**
Add a system property which allows you to bypass the RPC layer of appengine.

Previously we would use the `JettyHttpProxy` to convert the Jetty Request/Response into RPC and enter through the `DelegateConnector.service(DelegateExchange)`. 

After this change you can use the `appengine.use.HttpConnector` system property to bypass this and directly register an `HttpConnector` to handle requests.

### CODE REVIEW

**Comment by ludoch** [comment]
Trying to see how this PR will show up in google3. Does not mean it is approved, will need more chats.

**Review by gregw — DISMISSED** [review]
I need a bit better javadoc to fully review this. Can you add at least javadoc to all new interfaces and classes.

File: runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/http/JettyHttpHandler.java [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+  @Override
+  public boolean handle(Request request, Response response, Callback callback) throws Exception {
+    JettyRequestAPIData genericRequest = new JettyRequestAPIData(request, appInfoFactory, passThroughPrivateHeaders);
+    JettyResponseAPIData genericResponse = new JettyResponseAPIData(response);
+
+    // Read time remaining in request from headers and pass value to LocalRpcContext for use in
+    // reporting remaining time until deadline for API calls (see b/154745969)
+    Duration timeRemaining = genericRequest.getTimeRemaining();
+
+    // TODO: Can we get rid of this? or do we need to implement MessageLite?
```
Comment (ludoch) [inline runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/http/JettyHttpHandler.java:94]: Safe for now?

File: runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/http/JettyHttpHandler.java [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(appVersion.getClassLoader());
+    try {
+      runnable.run();
+    } finally {
+      Thread.currentThread().setContextClassLoader(oldClassLoader);
+    }
+
+    /*
+      todo we don't really need this because jetty response has these values
```
Comment (ludoch) [inline runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/http/JettyHttpHandler.java:188]: Safe for now? Any thing we still need to do?

**Comment by ludoch** [comment]
In general, try to avoid reformatting with Webtide guidelines for import orders, this conflicts with google style guide...
Soon, we should force the google guidelines plugin to this repository.
Not critical today, but we will do a refactor after.

**Comment by lachlan-roberts** [comment]
> In general, try to avoid reformatting with Webtide guidelines for import orders, this conflicts with google style guide...

I opened this issue for this https://github.com/GoogleCloudPlatform/appengine-java-standard/issues/111. 
I am not using the Webtide style for these PRs, I think just the intellij default.

**Review by ludoch — APPROVED** [review]

**Pull request closed without merging**